static/src/javascripts/projects/journalism/services/impl/viewport-scroll.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+
+let listening = false;
+let elementCount = 0;
+const elements = Object.create(null);
+
+const onScroll = () => {
+    const viewportHeight = window.innerHeight;
+
+    Object.keys(elements).forEach((threshold: string) => {
+        elements[threshold].forEach(({ element, callback }) => {
+            const rect = element.getBoundingClientRect();
+            const isNotHidden =
+                rect.top + rect.left + rect.right + rect.bottom !== 0;
+            const area = (rect.bottom - rect.top) * (rect.right - rect.left);
+            const visibleAreaPrime =
+                rect.top >= viewportHeight
+                    ? 0
+                    : (Math.min(viewportHeight, rect.bottom) -
+                          Math.max(0, rect.top)) *
+                      (rect.right - rect.left);
+            const visibleArea = rect.bottom <= 0 ? 0 : visibleAreaPrime;
+            const intersectionRatio = visibleArea / area;
+            if (isNotHidden && intersectionRatio >= parseInt(threshold, 10)) {
+                setTimeout(callback, 0, visibleArea);
+            }
+        });
+    });
+};
+
+const observe = (element: Element, threshold: number, callback: () => void) => {
+    if (!listening) {
+        window.addEventListener('scroll', onScroll);
```

Comment:
i'm really not happy with adding _another_ scroll listener for _another_ feature on the site. we already struggle with chucking everything we can think of at the screen. can this not use the throttledscroll?

Reply:
oh yeah of course, I thought I would do it but then forgot. Thanks!